### PR TITLE
[HWORKS-1061] Separate out of memory rules for ndbmtd nodes

### PR DESCRIPTION
**Pull request by kouzant**
[HWORKS-1061] Attribute to explicitly mark a node_exporter as ndbmtd node

[HWORKS-1061]: https://hopsworks.atlassian.net/browse/HWORKS-1061?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: templates/default/rules/host.rules.yml.erb [hunk 1]
```diff
@@ -2,13 +2,22 @@ groups:
...
     for: 5m
     labels:
       type: system-alert
     annotations:
       summary: "Out of memory (instance {{ $labels.instance }})"
-      description: "Node memory is filling up (< 10% left)\n  VALUE = {{ $value }}\n  LABELS: {{ $labels }}"
+      description: "Node memory is filling up (< <%= node['prometheus']['rules']['all-hosts-available-mem-threshold'] %>% left)\n  VALUE = {{ $value }}\n  LABELS: {{ $labels }}"
+
+  - alert: RonDBDatanodesOutOfMemory
+    expr: node_memory_MemAvailable_bytes{tags != ".*ndbmtd.*"} / node_memory_MemTotal_bytes{tags != ".*ndbmtd.*"} * 100 < <%= node['prometheus']['rules']['ndbd-hosts-available-mem-threshold'] %>
```
Comment (maismail) [inline templates/default/rules/host.rules.yml.erb:14]: shouldn't this be `=~` instead?

File: templates/default/node-exporter-consul.hcl.erb [hunk 2]
```diff
@@ -2,7 +2,10 @@ service {
     id = "node_exporter/monitoring"
     name = "node_exporter"
     tags = [
-        "monitoring"
+        "monitoring",
```
Comment (maismail) [inline templates/default/node-exporter-consul.hcl.erb:5]: does hcl tolerate commas at the end is no more values?

**Review by maismail — CHANGES_REQUESTED** [review]

File: templates/default/rules/host.rules.yml.erb (hunk 1 above)
Comment (kouzant) [reply templates/default/rules/host.rules.yml.erb:14]: Yes, you're right

File: templates/default/node-exporter-consul.hcl.erb (hunk 2 above)
Comment (kouzant) [reply templates/default/node-exporter-consul.hcl.erb:5]: Yes it does

File: attributes/default.rb [hunk 3]
```diff
@@ -107,3 +108,7 @@
 default['hopsmonitor']['prometheus']['kube-crt']         = ""
 default['hopsmonitor']['prometheus']['kube-key']         = ""
 default['hopsmonitor']['prometheus']['kube-ca']          = ""
+
+# Configuration attributes for alerting rules
+default['prometheus']['rules']['all-hosts-available-mem-threshold']  = "10"
+default['prometheus']['rules']['ndbd-hosts-available-mem-threshold'] = "3"
```
Comment (maismail) [inline attributes/default.rb:114]: btw, for app we set this to `1.5%` is `3%` a realistic default?

Comment (kouzant) [reply attributes/default.rb]: Well that's something very specific to the machine's specs. A VM with 250G of RAM with the default RonDB memory configuration leaves 4G for OS. I changed this to `1.5%`

**Review by maismail — APPROVED** [review]

**Pull request merged**